internal/corazawaf/body_buffer.go

```diff
@@ -50,8 +51,24 @@ func (br *BodyBuffer) Write(data []byte) (n int, err error) {
 		return 0, nil
 	}
 
-	l := int64(len(data)) + br.length
-	if l > br.options.MemoryLimit {
+	var targetLen int64
+	// Overflow check
+	if br.length == math.MaxInt64 || br.length >= (math.MaxInt64-int64(len(data))) {
```

Comment:
A thought came to my mind and can be addressed in a new PR. Do we really want to support payloads bigger than MaxInt? While theoretically this is correct in practice I don't think such payloads will exist and if they do we won't be able to read them and actually we can fill disc if we week ingesting data to disc (up to 32GB apparently) which we don't want to. I would say on max int we just fail and not keep writing. What does modsec does?